executor/calibrate_resource.go

```diff
@@ -83,13 +93,78 @@ type baseResourceCost struct {
 	writeReqCount uint64
 }
 
+const lowUsageThreshold = 10.
+const percentOfPass = 0.9
+const discardRate = 0.1
+
 type calibrateResourceExec struct {
 	baseExecutor
-
+	optionList   []*ast.DynamicCalibrateResourceOption
 	workloadType ast.CalibrateResourceType
 	done         bool
 }
 
+func (e *calibrateResourceExec) checkDynamicCalibrateOptions() (string, string, error) {
+	var startTime, endTime time.Time
+	var startTimeStr, endTimeStr string
+	checkDurationFn := func() error {
+		// check the duration
+		duration := endTime.Sub(startTime)
+		if duration > time.Hour*24 {
+			return errors.Errorf("the duration of calibration is too long.")
+		}
+		if duration < time.Minute*10 {
+			return errors.Errorf("the duration of calibration is too short.")
+		}
+		return nil
+	}
+	if len(e.optionList) == 2 {
```

Comment:
I'd like to replace this if-else with something like following:
```
var start, end, dur *ptr
for _, op := range e.optionList[0] {
  ...
}
if duration == nil { ...default for duration}
if start == nil { ...default-for-start }
if end == nil { ...default-for-end }

validate_start_end_duration()

...rest logics
```

And Since The static and dynamic branch have little common logic with each other, please wrap both of them with a separate function to avoid too long if..else.. block